File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_map<int,int> has;
        int start;
        int max_len = 0;
        int curr_len = 0;
        for(int s:nums){
            has[s] += has[s] + 1;
        }

        for(int e:nums){

            if(has.find(e-1)==has.end()){
                start = e;
                curr_len = 1;
                while(has.find(start+1)!=has.end()){
                    curr_len += 1;
                    start = start+1;
                }
            }

            max_len = max(max_len,curr_len);
        }

        return max_len;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        left = 0
        right = 0
        has = {}
        curr_len = 0
        max_len = 0
        while(left<=right and right<len(s)):
            if s[right] not in has:
                has[s[right]] = 1
                right += 1
            else:
                
                while s[right] in has:
                    del has[s[left]]
                    left += 1
            max_len = max(max_len, len(has.keys()))     
        return max_len
            

        